script.sublissimo/resources/lib/player_controller.py

```diff
@@ -0,0 +1,42 @@
+import xbmc
+
+from .play_along_file import PlayAlongFile
+from .sync_wizard import SyncWizard
+from .sync_by_frame_rate import SyncWizardFrameRate
+
+class SearchFrameRate(xbmc.Player):
+    def __init__ (self):
+        xbmc.Player.__init__(self)
+        self.frame_rate = 25
+
+    def get_frame_rate(self):
+        xbmc.sleep(1000)
+        self.frame_rate = xbmc.getInfoLabel('Player.Process(VideoFPS)')
+        self.stop()
+        return float(self.frame_rate)
+
+    def get_frame_rate_from_playing_file(self):
+        self.frame_rate = xbmc.getInfoLabel('Player.Process(VideoFPS)')
+        return float(self.frame_rate)
+
+class PlayerInstance:
+    def __init__(self):
+        self.instances = []
+
+    def __new__(cls):
```

Comment:
Is this trying to define a singleton? I don't quite see what's the use for `cls.instance.instances = []`. Seems to me it'll always be an empty list isn't it?

Reply:
Yeah, it suppossed to be a Singleton. In the past there were problems with menu's showing on top of each other because they were part of separate Player instances. I thought a single point of creation would be more clear and would make trouble shooting in the future more easy...  The cls.instance.instances = [] should have been deleted of course.